python/cuml/dask/ensemble/randomforestcommon.py

```diff
@@ -0,0 +1,173 @@
+import math
+
+from cuml.dask.common import raise_exception_from_futures
+from cuml.dask.common.input_utils import DistributedDataHandler, \
+    concatenate
+
+from dask.distributed import wait
+
+from uuid import uuid1
+
+
+class BaseRandomForestModel(object):
+
+    def _create_the_model(self, model_func, **kwargs):
+        n_workers = len(self.workers)
+        if self.n_estimators < n_workers:
+            raise ValueError(
+                "n_estimators cannot be lower than number of dask workers."
+            )
+
+        n_est_per_worker = math.floor(self.n_estimators / n_workers)
+
+        for i in range(n_workers):
+            self.n_estimators_per_worker.append(n_est_per_worker)
+
+        remaining_est = self.n_estimators - (n_est_per_worker * n_workers)
+
+        for i in range(remaining_est):
+            self.n_estimators_per_worker[i] = (
+                self.n_estimators_per_worker[i] + 1
+            )
+        seeds = list()
+        seeds.append(0)
+        for i in range(1, len(self.n_estimators_per_worker)):
+            sd = self.n_estimators_per_worker[i-1] + seeds[i-1]
+            seeds.append(sd)
+
+        key = str(uuid1())
+        self.rfs = {
+            worker: self.client.submit(
+                model_func,
+                n_estimators=self.n_estimators_per_worker[n],
+                seed=seeds[n],
+                **kwargs,
+                key="%s-%s" % (key, n),
+                workers=[worker],
+            )
+            for n, worker in enumerate(self.workers)
+        }
+
+        rfs_wait = list()
+        for r in self.rfs.values():
+            rfs_wait.append(r)

```

Comment:
Could we replace 51-55 with just `wait(self.rfs.values()))` or maybe `wait(list(self.rfs.values()))`?